src/Nox.ClientApi.Tests/Tests/Application/Messaging/DomainEventTests.cs

```diff
@@ -0,0 +1,47 @@
+using FluentAssertions;
+using ClientApi.Application.Dto;
+using AutoFixture;
+using Nox.Types;
+using Xunit.Abstractions;
+using ClientApi.Application.IntegrationEvents.StoreOwner;
+using ClientApi.Application.IntegrationEvents;
+
+namespace ClientApi.Tests.Application.Messaging
+{
+    [Collection("Sequential")]
+    public class DomainEventTests : NoxWebApiTestBase
+    {
+        private const string StoreOwnersControllerName = "api/storeowners";
+        private const string CountriesControllerName = "api/countries";
+
+        public DomainEventTests(
+            ITestOutputHelper testOutput,
+           NoxTestContainerService containerService)
+           : base(testOutput, containerService, true)
+        {
+        }
+
+        #region Domain Events
+
+        [Fact]
+        public async Task Post_Country_DomainEvent_SendsIntegrationEvent()
+        {
+            // Arrange
+            var createDto = new CountryCreateDto
+            {
+                Name = _fixture.Create<string>(),
+                Population = 100_000_001,
+            };
+
+            // Act
+            var result = await PostAsync<CountryCreateDto, CountryDto>(CountriesControllerName, createDto);
+
+            //Assert
+            result.Should().NotBeNull();
+
+            (await MassTransitTestHarness.Published.Any<Nox.Messaging.NoxMessageRecord<CountryPopulationHigherThan100M>>()).Should().BeTrue();
```

Comment:
You should be testing the CountryCreated event and not the Integration Event.

Reply:
Also, please add a test for the other events like Updated, and Deleted.

Reply:
this can be a little bit challenging because we are doing System.Testing here, and mocking the IMediator, a core component of Nox,  and preserving the behavior can be difficult

Maybe we can have a different approach, and do more "business test",  for example build a scenario where when a country is created with a certain property value in the client application you subscribe to the countrycreated domain event and you update some property other of the country, the you can do a getbyid from the api for that country and assert that is the expected value .... so basically the DomainEvent was raised, the Notification handler was called and the bossiness rule was executed

@gjorgji-grgovski-authoritypartners what do you think?